### ISSUE TITLE
The daemon did not get the correct service subnet

### ISSUE THREAD

**Issue by johnlanni** [telepresenceio/telepresence#1840]
When I was using telepresence intercept, I found that it is possible to access pods' IPs locally, but not to access some services' IPs. I saw this from the daemon.log:

![image](https://user-images.githubusercontent.com/6763318/124596383-c8c7de80-de94-11eb-9459-71376561d39a.png)

Only added this service subnet: `10.96.0.0/16`

Then I read the code in `cmd/traffic/cmd/manager/internal/cluster/info.go` and found that the way to obtain the service CIDR is like this: https://stackoverflow.com/a/61685899

However, the result of running the below command locally is: `10.96.0.0/14`

```bash
echo '{"apiVersion":"v1","kind":"Service","metadata":{"name":"tst"},"spec":{"clusterIP":"1.1.1.1","ports":[{"port":443}]}}' | kubectl apply -f - 2>&1 |sed 's/.*valid IPs is //'
```

Why the subnet is not correct ?

**Comment by johnlanni** [comment]
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: "2021-07-08T07:00:16Z"
  managedFields:
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:rules: {}
    manager: telepresence
    operation: Update
    time: "2021-07-08T07:00:16Z"
  name: traffic-manager-ambassador
  resourceVersion: "113565214"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/traffic-manager-ambassador
  uid: 6d5a4e84-f003-4615-a162-1e001f7e47a8
rules:
- apiGroups:
  - ""
  resources:
  - services
  verbs:
  - get
  - list
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - list
  - get
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - list
  - get
  - watch
```

This is the ClusterRole which subject to traffic-manager.
It seems a bug, traffic-manager does not have permission to create the bogus service.

**Comment by josecv** [comment]
@johnlanni in addition to the clusterrole, the traffic-manager should have a `Role` in the `ambassador` namespace that allows it to create that bogus service -- if you do `kubectl get role -n ambassador`, do you see a `traffic-manager` role? I'm not super sure that what you're seeing is down to not being able to create the dummy service, I think in that case we wouldn't see the service CIDR being parsed at all.

If you have the time, could you try your `echo` command in-cluster? You should even be able to `kubectl exec -ti` into the `traffic-manager` pod:

```bash
kubectl get pod -n ambassador # Take note of the traffic manager pod's name

kubectl exec -ti -n ambassador $TRAFFIC_MANAGER_POD

# Once inside the pod:
apk add curl
curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl
chmod a+x kubectl
cp kubectl /usr/local/bin/
echo '{"apiVersion":"v1","kind":"Service","metadata":{"name":"tst"},"spec":{"clusterIP":"1.1.1.1","ports":[{"port":443}]}}' | kubectl apply -f - 2>&1 |sed 's/.*valid IPs is //'
```

It would be very interesting to validate a) whether the service can be created, and b) whether the in-cluster response is actually different from your local response (as opposed to telepresence just parsing it wrong)

**Comment by josecv** [comment]
@johnlanni disregard my previous (now deleted) comment -- it seems @thallgren beat me to the punch 😄 #1853 might be the fix.

Altering the `ClusterRole` shouldn't be necessary, as `telepresence` should have a `Role` in the `ambassador` namespace that allows it to create the dummy service -- it's just that for that to work, the dummy service needs to actually be created in the `ambassador` namespace 🙃 which is what #1853 will do

**Comment by thallgren** [comment]
Closing this. Fix was delivered in 2.3.4.
